applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_scan_for_conn.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_mgmt_scan_for_conn_internal.h"
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+
+#include "bt_mgmt.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(bt_mgmt);
+
+#define CONNECTION_PARAMETERS                                                                      \
+	BT_LE_CONN_PARAM(CONFIG_BLE_ACL_CONN_INTERVAL, CONFIG_BLE_ACL_CONN_INTERVAL,               \
+			 CONFIG_BLE_ACL_SLAVE_LATENCY, CONFIG_BLE_ACL_SUP_TIMEOUT)
+
+static uint8_t bonded_num;
+struct bt_le_scan_cb scan_callback;
+static bool cb_registered;
+static char const *srch_name;
+
+static void bond_check(const struct bt_bond_info *info, void *user_data)
+{
+	char addr_buf[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(&info->addr, addr_buf, BT_ADDR_LE_STR_LEN);
+
+	LOG_DBG("Stored bonding found: %s", addr_buf);
+	bonded_num++;
+}
+
+static void bond_connect(const struct bt_bond_info *info, void *user_data)
+{
+	int ret;
+	const bt_addr_le_t *adv_addr = user_data;
+	struct bt_conn *conn;
+
+	if (!bt_addr_le_cmp(&info->addr, adv_addr)) {
+		LOG_DBG("Found bonded device");
+
+		bt_le_scan_cb_unregister(&scan_callback);
+		cb_registered = false;
+
+		ret = bt_le_scan_stop();
+		if (ret) {
+			LOG_WRN("Stop scan failed: %d", ret);
+		}
+
+		ret = bt_conn_le_create(adv_addr, BT_CONN_LE_CREATE_CONN, CONNECTION_PARAMETERS,
+					&conn);
+		if (ret) {
+			LOG_WRN("Create ACL connection failed: %d", ret);
+
+			ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_CONN, NULL);
+			if (ret) {
+				LOG_WRN("Failed to restart scanning: %d", ret);
+			}
+		}
+	}
+}
+
+/**
+ * @brief	Check advertising data for matching device name
+ *
+ * @param	data		The advertising data to check
+ * @param	user_data	The user data will contain pointer to the address
+ *
+ * @return	false to signify that we should stop going through adv data, true to continue
+ */
+static bool device_name_check(struct bt_data *data, void *user_data)
+{
+	int ret;
+	bt_addr_le_t *addr = user_data;
+	struct bt_conn *conn;
+
+	/* Check type == COMPLETE_NAME */
+	if (data->type != BT_DATA_NAME_COMPLETE) {
+		return true;
+	}
+
+	if ((data->data_len == strlen(srch_name)) &&
+	    (strncmp(srch_name, data->data, strlen(srch_name)) == 0)) {
+		LOG_DBG("Device found: %s", srch_name);
+
+		bt_le_scan_cb_unregister(&scan_callback);
+		cb_registered = false;
+
+		ret = bt_le_scan_stop();
+		if (ret) {
+			LOG_WRN("Stop scan failed: %d", ret);
+		}
+
+		ret = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, CONNECTION_PARAMETERS, &conn);
+		if (ret) {
+			LOG_ERR("Could not init connection");
+
+			ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_CONN, NULL);
+			if (ret) {
+				LOG_WRN("Failed to restart scanning: %d", ret);
+			}
+		}
+
+		return false;
+	}
+
+	return true;
```

Comment:
But returning `true` here you assume that the device is advertising multiple names, and that you hope that one of the other names fits `srch_name`. Is that intentional?

Reply:
Good point, changed so that we only check `if (data->data_len == strlen...` if `data->type == BT_DATA_NAME_COMPLETE`, so that if the type is not name we return true directly